File: PHP/ChainOfResponsibility/COR1/DogHandler.php
<?php


namespace TheRayCode\ChainOfResponsibility;


class DogHandler extends AbstractHandler
{

    public function handle(string $request): ?string
    {
        if ($request === "Bone") {
            return "Dog: I'll eat the " . $request . ".<br/>";
        } else {
            return parent::handle($request);
        }
    }
}

File: Csharp/Behavioral/ChainOfResponsibility/Example/ChainOfResponsibility/DogHandler.cs
namespace ChainOfResponsibility
{
    class DogHandler : AbstractHandler
    {
        public override object Handle(object request)
        {
            if (request.ToString() == "Bone")
            {
                return $"Dog: Oh my!! I'll eat the {request.ToString()}.\n";
            }
            else
            {
                return base.Handle(request);
            }
        }
    }
}